### PR TITLE
remove single-file-build.js, add build.js based on rollupjs

### PR DESCRIPTION
**Pull request by rossipedia**
Addresses #1009 

Currently this depends on a branch I have (PR submitted) for rollup-plugin-typescript2 which enables .d.ts support, so that it can be tested and verified until the PR is merged.

`npm test` passes

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -7,19 +7,21 @@
...
-		]
+        "strictNullChecks": true,
+        "noImplicitReturns": true,
+        "noImplicitUseStrict": true,
+        "noImplicitAny": false,
+        "moduleResolution": "node",
+        "lib": [
+            "es5",
+            "dom",
+			"es6"
```
Comment (rossipedia) [inline tsconfig.json]: hmmm not sure how that slipped in, will fix

**Comment by rossipedia** [comment]
Ahhh ok, looks like it doesn't like `async` in build.js, will fix that up

**Comment by rossipedia** [comment]
Hmmmm looks like `rollup-plugin-typescript2` depends on `@types/node` which is causing the conflict. Best route is probably just have mobx depend on that directly

**Comment by rossipedia** [comment]
Nailing down a couple odd errors (something's complaining about `setPrototypeOf`), need to take a quick break, tho. Will just back on this in a few.

**Comment by rossipedia** [comment]
Ok, went another way completely, but I think it's better. See 8dbc35a594 commit message for details

`npm run full-test` passes.

**Comment by rossipedia** [comment]
Oh! I totally forgot. Now with rollup.js it should be fairly trivial to generate a `mobx.mjs` file that can be referenced in `package.module` for consumers that want to tree-shake mobx.

**Comment by mweststrate** [comment]
Doublecheck: this one is blocked until  rollup-plugin-typescript2 is merged?

**Comment by rossipedia** [comment]
No. My latest commit removed that dependency. It's now just rollup (plus the filesize and progress plugins). Test suites pass and Travis is happy, it's good to merge if you're happy with it :)

**Comment by rossipedia** [comment]
(apologies for the spam). I also have another PR ready to go once this is accepted that adds an ES modules build (for tree shaking consumers).

**Comment by mweststrate** [comment]
Supercool! Hope to find some time tomorrow to merge!

Op di 30 mei 2017 om 19:49 schreef Bryan J. Ross <notifications@github.com>:

**Comment by mweststrate** [comment]
Tested and works great. Thanks!

**Pull request merged**